music_assistant/server/providers/siriusxm/__init__.py

```diff
@@ -0,0 +1,437 @@
+                yield self._parse_radio(channel)
+
+        if sub_path == "categories":
+            for category_key, category in self._categories.items():
+                if not category.name.endswith('Decade') and not category.name.endswith('Decades'):
+                    yield BrowseFolder(
+                        item_id=category_key,
+                        provider=self.domain,
+                        path=path + "/" + category_key,
+                        name=category.name,
+                        label="",
+                    )
+
+        if sub_path.startswith("categories/"):
+            category_key = sub_sub_path
+            channels = self._channels_by_category[category_key]
+            for channel in channels:
+                yield self._parse_radio(channel)
+
+    async def get_library_radios(self) -> AsyncGenerator[Radio, None]:
+        """Retrieve library/subscribed playlists from the provider."""
+        for channel_guid, channel in self._channels_by_id.items():
+            yield self._parse_radio(channel)
+
+    async def get_radio(self, channel_id: str) -> Union[Radio, None]:
+        channel = self._channels_by_id[channel_id] if channel_id in self._channels_by_id else None
+        if channel is None:
+            return None
+        return self._parse_radio(channel)
+
+    async def get_stream_details(self, item_id: str) -> StreamDetails:
+        """Get StreamDetails for a radio station."""
+        # channel_id = item_id
+        # channel = self._channels_by_id[channel_id]
+
+        # playlist = await self._client.get_playlist(channel.id, use_cache=False)
+        # playlist_items = list(filter(lambda l: not l.startswith("#"), playlist.split('\n')))
+
+        # first_segment = await self._client.get_segment(playlist_items[0])
+
+        # primary_hls_root = await self._client.get_primary_hls_root()
+        # first_item = f"{primary_hls_root}/{playlist_items[0]}"
+
+        # live_channel_raw = await self._client.get_now_playing(channel)
+        # live_channel = XMLiveChannel.from_dict(live_channel_raw["moduleList"]["modules"][0])
+
+        # playlist_url = live_channel.primary_hls.url
+
+        # cache_key = f"{self.instance_id}.media_info.{channel_id}"
+        # cached_info = await self.mass.cache.get(cache_key)
+        # if cached_info:
+        #     media_info = AudioTags.parse(cached_info)
+        # else:
+        #     # parse info with ffprobe (and store in cache)
+        #     media_info = await parse_tags(self._to_generator(first_segment))
+        #     await self.mass.cache.set(cache_key, media_info.raw)
+
+        # MPEG AAC Audio (mp4a)
+        # Type: Audio
+        # Channels: 2
+        # Sample Rate: 44100
+        # Bits per sample: 32
+
+        return StreamDetails(
+            provider=self.instance_id,
+            item_id=item_id,
+            audio_format=AudioFormat(
+                sample_rate=44100,
+                channels=2,
+                bit_depth=32,

```

Comment:
it is very, very unlikely that the bit_depth is 32 bits (that is only used in recording studios)
I think you meant 16 bits here ?

Reply:
Yeah I thought the same thing. I pulled this from VLC media/codec info. I had never seen 32-bit + 44.1KHz in the wild before. Usually it would be like 96KHz or 192KHz when people are going overboard on bit depth.

Reply:
well, you would hear mangled audio if its set to the wrong bit depth so you will know soon enough :-)